android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,70 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(eventSource: EventSource, customHandler: Handler? = null) : Application.ActivityLifecycleCallbacks {
+    private val CHECK_DELAY: Long = 800
+
+    private val eventSource = eventSource
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private var handler: Handler? = customHandler
+    private var paused = false
+    private var foreground = true
+    private var check: Runnable? = null
+
+    init {
+        if (handler == null) {
+            listenerThread.start()
+            handler = Handler(listenerThread.looper)
+        }
+    }
+    override fun onActivityResumed(activity: Activity) {
+        paused = false
+        if (check != null) {
+            handler!!.removeCallbacks(check!!) // stop check from running if the timer is still running
+            check = null
+        }
+        if (!foreground) {
+            Timber.d("Restarting connection")
+            eventSource.start()
+        }
+        foreground = true
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        paused = true
+
+        if (check != null) {
+            handler!!.removeCallbacks(check!!)
+            check = null
+        }
+
+        check = Runnable {
+            if (foreground && paused) {
+                foreground = false
+                Timber.d("App backgrounded for $CHECK_DELAY ms, closing connection")
+                eventSource.close()
+            } else {
+                Timber.d("App paused but still foreground")
```

Comment:
what does it mean for the app to be "paused" in the foreground?

Reply:
the onPause handler gets called if the user gets a phone call or swipes down to preview a notification, or if a dialog pops up, or if they go into multi-window mode and swipe over to another app, but in all those cases the app is still visible so we're considering it the foreground

Reply:
oh sike i took another read through and this check doesn't actually accomplish that, it's redundant to have two flags here